File: native_core_video_player/ios/Classes/NativeCoreVideoPlayerPlugin.swift
import Flutter
import UIKit

public class NativeCoreVideoPlayerPlugin: NSObject, FlutterPlugin {
    private var methodChannel: FlutterMethodChannel?
    private var eventChannel: FlutterEventChannel?
    private var eventSink: FlutterEventSink?
    private var textureRegistry: FlutterTextureRegistry?
    private var videoPlayer: VideoPlayer?
    
    public static func register(with registrar: FlutterPluginRegistrar) {
        let methodChannel = FlutterMethodChannel(name: "native_core_video_player", binaryMessenger: registrar.messenger())
        let eventChannel = FlutterEventChannel(name: "native_core_video_player/events", binaryMessenger: registrar.messenger())
        
        let instance = NativeCoreVideoPlayerPlugin()
        instance.methodChannel = methodChannel
        instance.eventChannel = eventChannel
        instance.textureRegistry = registrar.textures()
        
        registrar.addMethodCallDelegate(instance, channel: methodChannel)
        eventChannel.setStreamHandler(instance)
    }
    
    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "initialize":
            guard let args = call.arguments as? [String: Any],
                  let source = args["source"] as? String,
                  let textureRegistry = textureRegistry else {
                result(FlutterError(code: "INVALID_ARGUMENT", message: "Source is required", details: nil))
                return
            }
            
            videoPlayer?.dispose()
            videoPlayer = VideoPlayer(textureRegistry: textureRegistry) { [weak self] event in
                self?.eventSink?(event)
            }
            
            videoPlayer?.initialize(source: source)
            result(videoPlayer?.getTextureId())
            
        case "play":
            videoPlayer?.play()
            result(nil)
            
        case "pause":
            videoPlayer?.pause()
            result(nil)
            
        case "seekTo":
            guard let args = call.arguments as? [String: Any],
                  let position = args["position"] as? Int64 else {
                result(FlutterError(code: "INVALID_ARGUMENT", message: "Position is required", details: nil))
                return
            }
            videoPlayer?.seekTo(position: position)
            result(nil)
            
        case "setVolume":
            guard let args = call.arguments as? [String: Any],
                  let volume = args["volume"] as? Double else {
                result(FlutterError(code: "INVALID_ARGUMENT", message: "Volume is required", details: nil))
                return
            }
            videoPlayer?.setVolume(volume: Float(volume))
            result(nil)
            
        case "setPlaybackSpeed":
            guard let args = call.arguments as? [String: Any],
                  let speed = args["speed"] as? Double else {
                result(FlutterError(code: "INVALID_ARGUMENT", message: "Speed is required", details: nil))
                return
            }
            videoPlayer?.setPlaybackSpeed(speed: Float(speed))
            result(nil)
            
        case "setLooping":
            guard let args = call.arguments as? [String: Any],
                  let looping = args["looping"] as? Bool else {
                result(FlutterError(code: "INVALID_ARGUMENT", message: "Looping is required", details: nil))
                return
            }
            videoPlayer?.setLooping(looping: looping)
            result(nil)
            
        case "dispose":
            videoPlayer?.dispose()
            videoPlayer = nil
            result(nil)
            
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}

extension NativeCoreVideoPlayerPlugin: FlutterStreamHandler {
    public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        return nil
    }
    
    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
        self.eventSink = nil
        return nil
    }
}

File: native_core_video_player/android/src/main/kotlin/com/example/native_core_video_player/NativeCoreVideoPlayerPlugin.kt
package com.example.native_core_video_player

import android.app.Activity
import android.content.Context
import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import io.flutter.view.TextureRegistry

class NativeCoreVideoPlayerPlugin: FlutterPlugin, MethodCallHandler, ActivityAware {
    private lateinit var methodChannel: MethodChannel
    private lateinit var eventChannel: EventChannel
    private var eventSink: EventChannel.EventSink? = null
    private lateinit var context: Context
    private lateinit var textureRegistry: TextureRegistry
    private var activity: Activity? = null
    private var videoPlayer: VideoPlayer? = null

    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        context = flutterPluginBinding.applicationContext
        textureRegistry = flutterPluginBinding.textureRegistry
        
        methodChannel = MethodChannel(flutterPluginBinding.binaryMessenger, "native_core_video_player")
        methodChannel.setMethodCallHandler(this)
        
        eventChannel = EventChannel(flutterPluginBinding.binaryMessenger, "native_core_video_player/events")
        eventChannel.setStreamHandler(object : EventChannel.StreamHandler {
            override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
                eventSink = events
            }

            override fun onCancel(arguments: Any?) {
                eventSink = null
            }
        })
    }

    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
        when (call.method) {
            "initialize" -> {
                val source = call.argument<String>("source")
                if (source == null) {
                    result.error("INVALID_ARGUMENT", "Source is required", null)
                    return
                }
                
                videoPlayer?.dispose()
                videoPlayer = VideoPlayer(context, textureRegistry) { event ->
                    eventSink?.success(event)
                }
                
                videoPlayer?.initialize(source)
                result.success(videoPlayer?.textureId)
            }
            "play" -> {
                videoPlayer?.play()
                result.success(null)
            }
            "pause" -> {
                videoPlayer?.pause()
                result.success(null)
            }
            "seekTo" -> {
                val position = call.argument<Number>("position")?.toLong()
                if (position == null) {
                    result.error("INVALID_ARGUMENT", "Position is required", null)
                    return
                }
                videoPlayer?.seekTo(position)
                result.success(null)
            }
            "setVolume" -> {
                val volume = call.argument<Number>("volume")?.toFloat()
                if (volume == null) {
                    result.error("INVALID_ARGUMENT", "Volume is required", null)
                    return
                }
                videoPlayer?.setVolume(volume)
                result.success(null)
            }
            "setPlaybackSpeed" -> {
                val speed = call.argument<Number>("speed")?.toFloat()
                if (speed == null) {
                    result.error("INVALID_ARGUMENT", "Speed is required", null)
                    return
                }
                videoPlayer?.setPlaybackSpeed(speed)
                result.success(null)
            }
            "setLooping" -> {
                val looping = call.argument<Boolean>("looping")
                if (looping == null) {
                    result.error("INVALID_ARGUMENT", "Looping is required", null)
                    return
                }
                videoPlayer?.setLooping(looping)
                result.success(null)
            }
            "dispose" -> {
                videoPlayer?.dispose()
                videoPlayer = null
                result.success(null)
            }
            else -> {
                result.notImplemented()
            }
        }
    }

    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
        methodChannel.setMethodCallHandler(null)
        eventChannel.setStreamHandler(null)
        videoPlayer?.dispose()
        videoPlayer = null
    }

    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
        activity = binding.activity
    }

    override fun onDetachedFromActivityForConfigChanges() {
        activity = null
    }

    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
        activity = binding.activity
    }

    override fun onDetachedFromActivity() {
        activity = null
        videoPlayer?.pause()
    }
}
